### PR TITLE
Add script to create better africa.geojson

### PR DESCRIPTION
**Pull request by ryaanahmed**
This PR adds a data preprocessing script to construct a higher resolution GeoJSON file, which includes island countries missing from our earlier JSON.

A few things left to be done here -- particularly better handling of disputed territories.

This also creates a situation in the map quiz where we now have all of the countries... but the geometries for, e.g., Cape Verde are too small to click on easily. Not impossible, but hard to use. We'll probably have to use the bounding boxes instead of the paths for the click handlers for those.

### CODE REVIEW

File: backend/data/create_africa_geojson.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    # sort by name
+    africa_features.sort(key=lambda feature: feature['properties']['name'])
+
+    # Add Western Saraha and Somaliland geometries
+    western_sahara_fid = 18
+    somaliland_fid = 19
+    for territory in disputed_features:
+        fid = territory['properties']['FID']
+        if fid in [western_sahara_fid, somaliland_fid]:
+            africa_features.append(territory)
```
Comment (JasonLin43212) [inline backend/data/create_africa_geojson.py]: Currently, the disputed territories do not have a name or ISO when they are sent to the front end. There are many ways in which we can accomplish that such as copying the jet punk map quiz and letting West Sahara be a part of Morocco and Somaliland be a Somalia. Another way can be to also add in those territories as extra countries. Although Somaliland is in the V-Dem data set, Western Sahara is not.

Comment (ryaanahmed) [reply backend/data/create_africa_geojson.py]: Better to get a temporary solution in now than to merge this as is.

I'm going to combine the geometries for Western Sahara + Morocco and Somaliland + Somalia for now. We can refine with Evan later on.

File: backend/app/tests.py [hunk 2]
```diff
@@ -114,9 +114,8 @@ def test_loading_democracy_data(self):
         for score_type in max_values:
             self.assertNotEqual(max_values[score_type], "")
 
-        # Change it to include other missing countries
         # Assures that all 54 countries are in the data
-        # self.assertEqual(54, len(democracy_data))
+        self.assertEqual(54, len(democracy_data))
```
Comment (JasonLin43212) [inline backend/app/tests.py:118]: Thanks for doing this!

File: backend/data/create_africa_geojson.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    Read the CSV file from our v-dem based dataset and return
+    a dict mapping ISO A3 codes to country names
+    """
+    csv_name_index = 0
+    csv_iso_code_index = 1
+    iso_name_dict = {}
+    filename = 'lieberman_afr_data.csv'
+    with open(filename, encoding='utf-8') as democracy_score_file:
+        reader = csv.reader(democracy_score_file, delimiter=',')
+        _ = next(reader)  # skip headers
```
Comment (JasonLin43212) [inline backend/data/create_africa_geojson.py:41]: I don't think that you need to bind `next(reader)` to anything.

**Review by JasonLin43212 — APPROVED** [review]
Amazing! The map looks so much better.

File: backend/data/create_africa_geojson.py (hunk 3 above)
Comment (ryaanahmed) [reply backend/data/create_africa_geojson.py:41]: Yep, that's right -- that said, my habit in these situations is to use the underscore to signal that I'm intentionally ignoring the return value of this statement.

**Comment by ryaanahmed** [comment]
@JasonLin43212 -- did some additional scripting to merge the geometries of the disputed territories into the nearby nations. Please take another quick look to sanity check, then I'll merge.

**Comment by JasonLin43212** [comment]
@ryaanahmed It seems like the `merge_geometries` function sort of works. There seems to be some remaining lines between the disputed territories. I'm not sure how to go about removing them since you used a function from a library to do the merging.
som

**Comment by ryaanahmed** [comment]
Nice catch -- I'd noticed this seam, too. I think it's because the geometries are not completely overlapping / do not completely share a boundary. I'm going to put this in the category of "good enough for right now" and file an issue to improve down the road. I think we're going to have to do some manual tweaking.

**Pull request merged**